MaxText/layers/linears.py

```diff
@@ -429,37 +436,64 @@ def generate_masks(self, top_k_indices):
       expert_mask = jax.nn.one_hot(top_k_indices, num_classes=self.num_experts, dtype=jnp.int32)
       reshaped_expert_mask = jnp.reshape(expert_mask, (batch_size, seq_len * self.num_experts_per_tok, self.num_experts))
       expert_token_count = jnp.cumsum(reshaped_expert_mask, axis=1)
-      expert_token_count = jnp.reshape(expert_token_count, ((batch_size, seq_len, self.num_experts_per_tok, self.num_experts)))
-      trunc_expert_mask = expert_mask * jnp.less_equal(expert_token_count, expert_capacity)
+      reshaped_expert_token_count = jnp.reshape(expert_token_count, ((batch_size, seq_len, self.num_experts_per_tok, self.num_experts)))
+      trunc_expert_mask = expert_mask * jnp.less_equal(reshaped_expert_token_count, expert_capacity_per_batch)
       updated_expert_mask = jnp.sum(trunc_expert_mask, axis=3)
-      return updated_expert_mask
+
+      # update weights for all experts
+      softmax_probs *= updated_expert_mask
+      weights = self.update_weights(softmax_probs, top_k_indices)
+
+      # shape of expert_token_position is (batch_size, seq_len * num_experts_per_tok, num_experts)
+      # calculate token position in expert capacity dimension
+      expert_token_position = reshaped_expert_mask * expert_token_count
+      reshaped_expert_token_position = jnp.reshape(expert_token_position, (batch_size, seq_len, self.num_experts_per_tok, self.num_experts))
```

Comment:
Same comment as above for reshaped* named - prefer consistentcy so I can tell if (seq_len * exp/tok) or (seq_len, exp/tok) just from the name - I like using *_seq_fused to refer to the former,e.g. expert_token_position_seq_fused

Reply:
Thanks! I like this `fused` suffix. Let me do update variable names and try to keep them consistent.